gui/editgamedialog.cpp

```diff
@@ -470,6 +471,20 @@ void EditGameDialog::open() {
 	_platformPopUp->setSelected(sel);
 }
 
+void EditGameDialog::close() {
+	OptionsDialog::close();
+
+	// Cleanup engine widgets before unloading its plugin
+	if (_engineOptions) {
+		// Switch back to Game tab before deleting the widget
+		_tabWidget->setActiveTab(0);
+		_tabWidget->removeWidget(_engineOptions);
```

Comment:
This might need modifying now that PR #5832 has been merged - `_engineOptions` is now in an intermediate container.